frontend/src/app/services/source/source.service.ts

```diff
@@ -40,21 +39,23 @@ export class SourceService {
     source: Source
   ): Observable<Source> {
     return new Observable<Source>((subscriber) => {
-      this.projectService.init(project_slug).subscribe((project) => {
-        this.http
-          .post<Source>(
-            environment.backend_url +
-              '/projects/' +
-              project.name +
-              '/extensions/modelsources/git/create/' +
-              model_slug,
-            source
-          )
-          .subscribe((new_source) => {
-            subscriber.next(new_source);
-            subscriber.complete();
-          });
-      });
+      this.projectService
+        .getProjectBySlug(project_slug)
+        .subscribe((project) => {
+          this.http
+            .post<Source>(
+              environment.backend_url +
+                '/projects/' +
+                project.name +
+                '/extensions/modelsources/git/create/' +
+                model_slug,
+              source
+            )
+            .subscribe((new_source) => {
+              subscriber.next(new_source);
+              subscriber.complete();
+            });
+        });
```

Comment:
[mergeMap](https://rxjs.dev/api/operators/mergeMap) would be a good solution, right? (says the rxjs decision tree ;))

Reply:
Actually a switchMap is more suitable (we want to interrupt the previous subscription when starting a new one), but these chained subscribes have to be replaced anyway.